src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/BookingService.kt

```diff
@@ -911,6 +911,48 @@ class BookingService(
     return success(arrivalEntity)
   }
 
+  @Transactional
+  fun createCas3Arrival(
+    user: UserEntity? = null,
+    booking: BookingEntity,
+    arrivalDate: LocalDate,
+    expectedDepartureDate: LocalDate,
+    notes: String?,
+    keyWorkerStaffCode: String?,
+  ) = validated<ArrivalEntity> {
+    if (booking.premises !is TemporaryAccommodationPremisesEntity) {
```

Comment:
It would probably be worth updating the original `createArrival` method to return an error if the booking is for a Temporary Accommodation premises to indicate that this is the preferred function.